packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -249,6 +260,86 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         return db.recoveryKeyExists(uid);
       },
     },
+    {
+      method: 'GET',
+      path: '/recoveryKey/storageHint',
+      options: {
+        ...RECOVERY_KEY_DOCS.RECOVERYKEY_HINT_GET,
+        // The storageHint can be (or will be) retrieved during password reset with account recovery key
+        // (after following password reset link) or when authenticated (from /session)
+        // TODO: verify if we can allow to auth strategies
+        auth: {
+          strategy: ['accountResetToken', 'sessionToken'],
+        },
+      },
+      handler: async function (request) {
+        log.begin('getRecoveryKeyHint', request);
+
+        const { uid } = request.auth.credentials;
+
+        await customs.checkAuthenticated(request, uid, 'getRecoveryKeyHint');
+
+        const { storageHint } = await db.getRecoveryKeyHint(uid);
+
+        recordSecurityEvent('account.recovery_key_hint_viewed', {
```

Comment:
@vbudhram Should we have security events for these? I wasn't sure if these should be recorded.